astroid/bases.py

```diff
@@ -128,7 +128,7 @@ def infer(
 def _infer_stmts(
     stmts: Sequence[nodes.NodeNG | type[Uninferable] | Instance],
     context: InferenceContext | None,
-    frame: nodes.NodeNG | Instance | None = None,
+    frame: nodes.NodeNG | Instance | str | None = None,
```

Comment:
```suggestion
    frame: nodes.NodeNG | Instance | None = None,
```
See comment regarding `self.lookup` below.